scene/gui/text_edit.cpp

```diff
@@ -4226,7 +4226,10 @@ double TextEdit::get_scroll_pos_for_line(int p_line, int p_wrap_index) const {
 		return p_line;
 	}
 
-	double new_line_scroll_pos = get_visible_line_count_in_range(0, CLAMP(p_line, 0, text.size() - 1));
+	double new_line_scroll_pos = 0.0;
+	if (p_line > 0) {
+		new_line_scroll_pos = get_visible_line_count_in_range(0, MIN(p_line - 1, text.size() - 1));
+	}
```

Comment:
Should now be able to do:
```suggestion
	double new_line_scroll_pos = get_visible_line_count_in_range(0, CLAMP(p_line, 0, text.size() - 1)) - 1;
```

Reply:
I think this doesn't work because `p_line`th line might be wrapped, increasing the return value by more than 1

Reply:
Ahh I see, good spot. 

Probably for the best to keep as is, otherwise we'd have to fix too many other places.